api/discord/controllers/discord.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const axios = require('axios');
+
+module.exports = {
+  async discordAuth(ctx){
+    const { code } = ctx.query;
+    if (code) {
+      try {
+        // post a request to get back the token type and an access token
+        const discordData = {
+          client_id: process.env.CLIENT_ID,
+          client_secret: process.env.CLIENT_SECRET,
+          code,
+          grant_type: 'authorization_code',
+          redirect_uri: process.env.REDIRECT_URL,
+          scope: 'identify',
+        };
+        const oauthResult = await axios.post(
+          'https://discord.com/api/oauth2/token',
+          new URLSearchParams(discordData),
+          {
+            headers: {
+              'Content-Type': 'application/x-www-form-urlencoded',
+            }
+          }
+        );
+
+        // second request to get back the user discord information
+        const userResult = await axios.get('https://discord.com/api/users/@me', {
+          headers: {
+            authorization: `${oauthResult.data.token_type} ${oauthResult.data.access_token}`,
+          },
+        });
+
+        // update the discord information to the user
+        const validatedBody = {
+          discordId: userResult.data.id,
+          discordUsername: userResult.data.username,
+          discordAvatar: userResult.data.avatar,
+          discordDiscriminator: userResult.data.discriminator,
+        };
+        const { id } = ctx.state.user;
+        const { username } = ctx.state.user;
+        await strapi.query('user', 'users-permissions').update({ id: id}, validatedBody);
```

Comment:
I see here that you're doing `{ id: id }` since the key and the variable name are the same you can just put the name of the variable, so instead of the way you were doing it you can do it this way.
```js
await strapi.query('user', 'users-permissions').update({ id }, validatedBody);
```

Reply:
Does { id } and { id: id } mean the same thing in this case？

Reply:
> Does { id } and { id: id } mean the same thing in this case？

Yes, you can only do this if the key name and the variable name your setting as the value are the same. If they are different, you'll have to specify the key and value separately. For example, 
```js
//different names
{id: userId}

//same name
{ id }
```